checker/manual/constant-value-checker.tex

```diff
@@ -185,22 +224,27 @@ \section{Warnings\label{value-checker-warnings}}
 \end{itemize}
 \end{sloppypar}
 
-There is one other situation in which the Constant Value Checker produces a
+There are some other situations in which the Constant Value Checker produces a
 warning message:
 
 \begin{sloppypar}
 \begin{itemize}
-\item \code{[too.many.values.given] Annotation ignored because the maximum
-  number of values tracked is 10.}
+\item \code{[too.many.values.given] The maximum number of arguments permitted is 10.}
 
   The Constant Value Checker only tracks up to 10 possible values for an
   expression.  If you write an annotation with more values than will be
-  tracked, the annotation is ignored.
+  tracked, the annotation is replaced with \<@IntRange> or \<@UnknownVal>.
 
-\end{itemize}
-\end{sloppypar}
+\item \code{[too.narrow.int.range] The smallest interval of integer range permitted is 10.}
```

Comment:
The reason for changing the annotation silently (from @IntRange to @IntVal when range encompasses 10 or fewer values, and vice versa) is that computational complexity can be reduced in this way. For example, there is no need to consider the subtyping when assigning a @IntRange to @IntVal (IsSubtype always false), because the number of possible values of @IntRange is always greater than that of @IntVal.

This is also the case when in the future we are implementing the value propagation involving both @IntVal and @IntRange. For example, if we are adding an integral annotated with @IntVal to an integral annotated with @IntRange, the result is always @IntRange because @IntRange annotated integral always has more than 10 possible values.